### PR TITLE
ptr - Create package to convert basic types to pointers

### PR DESCRIPTION
**Pull request by breerly**
@abhinav @nomis52 @prashantv @akshayjshah

### CODE REVIEW

File: ptr/ptr.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package ptr
```
Comment (prashantv) [inline ptr/ptr.go]: nit: can you add package docs here

File: ptr/ptr.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+// Byte converts an int8 to a pointer
+func Byte(x int8) *int8 {
```
Comment (abhinav) [inline ptr/ptr.go]: Rename to Int8 to match the others.

Comment (akshayjshah) [inline ptr/ptr.go]: This is a bit confusing - `byte` is an alias for `uint8`. This should either be `Int8` or take a `uint8`.

File: ptr/ptr.go (hunk 1 above)
Comment (breerly) [inline ptr/ptr.go]: Done.

**Comment by abhinav** [comment]
:shipit:

File: ptr/ptr.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	return &x
+}
+
```
Comment (prashantv) [inline ptr/ptr.go]: since we have all the int types except for `Int16`, should we add that as well?

File: ptr/ptr.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+// Int64 converts an int64 to a pointer
+func Int64(x int64) *int64 {
+	return &x
+}
+
+// Double converts a float64 to a pointer
+func Double(x float64) *float64 {
```
Comment (prashantv) [inline ptr/ptr.go]: @abhinav should we also have `Byte` to match `Double`? Or should this also be changed to `Float64`?

File: ptr/ptr.go (hunk 3 above)
Comment (breerly) [inline ptr/ptr.go]: Agreed, adding 👍  

Can you think of any others?

Comment (breerly) [inline ptr/ptr.go]: Added `Int16`.

**Comment by prashantv** [comment]
lgtm

File: ptr/ptr.go (hunk 4 above)
Comment (breerly) [inline ptr/ptr.go]: Changed to `Float64`.

**Pull request merged**